File: KnowledgeSharingApi/KnowledgeSharingApi.Domain/Models/ApiResponseModels/ResponseCourseCardModel.cs
﻿using KnowledgeSharingApi.Domains.Enums;
using KnowledgeSharingApi.Domains.Models.Entities;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace KnowledgeSharingApi.Domains.Models.ApiResponseModels
{
    public class ResponseCourseCardModel : Entity
    {
        public Guid UserItemId { get; set; }

        public string Title { get; set; } = string.Empty;

        public string? Abstract { get; set; } = string.Empty;

        public string? Thumbnail { get; set; }

        public EPrivacy Privacy { get; set; }

        //public string Introduction { get; set; } = string.Empty;

        public decimal Fee { get; set; }

        //public int EstimateTimeInMinutes { get; set; }

        public bool IsFree { get; set; }



        public ECourseRoleType? CourseRoleType { get; set; }

        public Guid? CourseRelationId { get; set; }



        public Guid UserId { get; set; }

        public string Username { get; set; } = string.Empty;

        public string FullName { get; set; } = string.Empty;

        public string? Avatar { get; set; } = null;

        protected override ResponseCourseCardModel Init()
        {
            return new ResponseCourseCardModel();
        }
    }
}

File: KnowledgeSharingVue/knowledge-sharing-vue/src/js/models/api-response-models/response-course-card-model.js

import Entity from '../entity.js';

class ResponseCourseCardModel extends Entity{
    constructor() {
        super();
        this.UserItemId = null;
        this.Title = null;
        this.Abstract = null;
        this.Thumbnail = null;
        this.Fee = null;
        this.IsFree = null;
        this.Privacy = null;
        this.Introduction = null;
        this.EstimateTimeInMinutes = null;

        this.CourseRoleType = null;
        this.CourseRelationId = null;

        this.UserId = null;
        this.Username = null;
        this.FullName = null;
        this.Abstract = null;
    }

    init() {
        return new ResponseCourseCardModel();
    }
}

export default ResponseCourseCardModel;

